### PR TITLE
[FriendlyUI only] add posts item ghost loading state

### PR DESCRIPTION
**Pull request by s-cheng**
This PR adds a ghost loading state for the post list item. Right now it's only used in `BookmarksList` but we'll add it to more places in subsequent PRs. A bunch of this is borrowed from the `service-worker-prototype` branch, so hopefully the merge is not too bad.

Desktop, light mode:

https://github.com/ForumMagnum/ForumMagnum/assets/9057804/c9c4a5d6-f424-4196-a653-9dec943c55b7

Mobile, dark mode:

https://github.com/ForumMagnum/ForumMagnum/assets/9057804/d068c515-dbcb-4f2a-b4d1-92d97365b113

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1207050154489980) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/common/LoadMore.tsx [hunk 1]
```diff
@@ -98,7 +98,7 @@ const LoadMore = ({
    * a non-zero count that graphql cached during SSR.
    */
   const isFirstRender = useIsFirstRender();
-  loading = loading && !(isFirstRender && (count ?? 0) > 0);
+  loading = loading && !isFirstRender;
```
Comment (s-cheng) [inline packages/lesswrong/components/common/LoadMore.tsx:101]: I feel like the `LoadMore` component should only show a loading animation after it's clicked, and not when it's first rendered. Let me know if you disagree.

File: packages/lesswrong/components/common/queryStatusUtils.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // Simple utility to check if a network status means the associated apollo query is updating
 //
-// https://github.com/apollographql/apollo-client/blob/master/packages/apollo-client/src/core/networkStatus.ts
+// https://github.com/apollographql/apollo-client/blob/main/src/core/networkStatus.ts
```
Comment (s-cheng) [inline packages/lesswrong/components/common/queryStatusUtils.ts:3]: This was out of date

**Comment by s-cheng** [comment]
@agnesstenlund FYI - I made a guess for the dark mode colors, and for the animation I just copied the one we already have for the user profile image to be consistent.

**Comment by agnesstenlund** [comment]
This looks great!! Very happy about this.

I think the animation is maybe a tad bit aggressive, that it's pulsating too intensely/quickly? Could also make the difference in colors less visible and the pulsating may appear less obvious (vs. changing the speed)

Seeing this live also makes me think all lines should actually be equal height (sorry for not realising that in the design stage)

**Comment by agnesstenlund** [comment]
another thing I'm thinking more when seeing it live is that for a split second I think the thing on the left is an avatar. Wdyt about doing one of these instead?
Screenshot 2024-04-11 at 14 51 46
Screenshot 2024-04-11 at 14 51 38

**Comment by s-cheng** [comment]
@agnesstenlund I've made the animation slower and the highlight colors more subtle, and made all the heights the same (except the three dots icon):

https://github.com/ForumMagnum/ForumMagnum/assets/9057804/11b71a51-1e70-4bc8-8d79-96575b90bc1f

https://github.com/ForumMagnum/ForumMagnum/assets/9057804/1e177fae-25e7-4bdd-a468-7e8917548078

**Comment by agnesstenlund** [comment]
I love it. Perfection

File: packages/lesswrong/components/common/LoadMore.tsx (hunk 1 above)
Comment (oetherington) [reply packages/lesswrong/components/common/LoadMore.tsx:101]: This seems kind-of scary to me, but I after a quick look I couldn't find anywhere that actually breaks so maybe it's fine - we should just keep an eye out for missing loaders in the future.

We should also update the comment above though as it's now out of date with the code.

**Review by oetherington — APPROVED** [review]
LGTM 👍

**Pull request merged**